src/analytics/formTracker/formTracker.ts

```diff
@@ -168,18 +168,36 @@ export class FormTracker extends BaseTracker {
     const field = document.getElementById(element.id);
     const fieldset = field?.closest("fieldset");
     const checkbox = element.type === "checkbox";
-    const radio = element.type === "radio buttons";
+    const radio = element.type === "radio";
+    const commonId = element.id.split("-")[0];
     if (fieldset) {
-      // If it's a child of a fieldset e.g radio button/ checkbox, look for the legend
+      // If it's a child of a fieldset ,look for the legend if not check for backup conditions
       const legendElement = fieldset.querySelector("legend");
       if (legendElement?.textContent) {
         return legendElement.textContent.trim();
       }
-      // if it is a checkbox or radio which does not have a legend, then check for the below conditions
+      const h1OrH2WithRel = document.querySelector(
```

Comment:
Tell me if I'm wrong but you've duplicated the same code we have below.
In this case, we need to extract it and put it in a function (with unit tests).

Reply:
Yes I am aware of code duplication , I mentioned in my description that there was, but the problem was if the element was in a fieldset but had no legend (screenshot of scenario in description), then for some reason the other the backup conditions weren't passing even if there was a h1 it would skip it and just return undefined. That's why I moved them inside that if block too. But yes I will put it in a function then unless there is a better way to cover the scenarios in the pr description

Reply:
Put in a function only the duplicated piece of code is the best solution.

Reply:
Ok a created a function within the function , to be used and replace the duplicate code. I can extract it out but I don't know if I need to ?  If its specific only to this function.